File: swift_client/Sources/EZAuth/Storage.swift
import Foundation

/// Storage usage.
public struct Storage: Sendable {
    let client: BaseClient

    public func tables() async throws -> StorageResponse {
        try await client.fetch(
            StorageResponse.self,
            path: "/v1/tables/storage",
            auth: .auto
        )
    }

    public func objects() async throws -> StorageResponse {
        try await client.fetch(
            StorageResponse.self,
            path: "/v1/buckets/storage",
            auth: .auto
        )
    }
}

// MARK: - Types

public struct StorageResponse: Decodable, Sendable {
    public let used_bytes: Int
    public let limit_bytes: Int
    public let used_percent: Double
}

File: python_client/src/ezauth_client/storage.py
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ._client import BaseClient


class Storage:
    def __init__(self, client: BaseClient):
        self._client = client

    def tables(self) -> dict:
        return self._client._fetch("/v1/tables/storage", auth="auto")

    def objects(self) -> dict:
        return self._client._fetch("/v1/buckets/storage", auth="auto")
